contracts/mocks/docs/account/MyFactoryAccount.sol

```diff
@@ -0,0 +1,47 @@
+// contracts/MyFactoryAccount.sol
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Clones} from "../../../proxy/Clones.sol";
+import {Address} from "../../../utils/Address.sol";
+
+/**
+ * @dev A factory contract to create accounts on demand.
+ */
+contract MyFactoryAccount {
+    using Clones for address;
+    using Address for address;
+
+    address private immutable _impl;
+
+    constructor(address impl_) {
+        _impl = impl_;
+    }
+
+    /// @dev Predict the address of the account
+    function predictAddress(bytes32 salt, bytes calldata callData) public view returns (address, bytes32) {
```

Comment:
I find it strange (and possibly confusing) to return the bytes calldataSalt in addition to the address.

IMO this function should return only the address. In `cloneAndInitialize`, if `predicted.code.length == 0` we can recompute the hash, that is not a big deal IMO.